### PR TITLE
Feature: find_apps support customer app class (RDT-589)

### PR DESCRIPTION
**Pull request by xd-ydchen**
I needs some specific features in my project:
- Building eco versions with different folder names
- Pre process magic sdkconfig files

I was able to achieve them in this way with this PR:
```python
class MyApp(CMakeApp):
    ECO_VERSION_PLACEHOLDER = '@e'  # replace it with the ECO version
    _eco: str

    def __init__(self, *args, **kwargs):
        if 'eco' not in kwargs:
            kwargs['eco'] = ''
        super().__init__(*args, **kwargs)

    def _initialize_hook(self, **kwargs):
        self._eco = kwargs['eco']

    def _expand(self, path: str) -> str:
        path = super()._expand(path)
        return path.replace(self.ECO_VERSION_PLACEHOLDER, self._eco)

    def _process_sdkconfig_files(self):
        import jinja2
        ...
        pass


apps = find_apps('xxx', build_system=MyApp, xxx)
apps = _filter_or_duplicate(apps)
build_apps(apps, xxxx)
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 All notable changes to this project will be documented in this file.
 
+## latest
```
Comment (hfudev) [inline CHANGELOG.md:5]: this project generates release notes from commit messages. No need to add manually.

File: idf_build_apps/finder.py [hunk 2]
```diff
@@ -29,7 +28,7 @@
 def _get_apps_from_path(
     path: str,
     target: str,
-    build_system: str = 'cmake',
+    app_cls: t.Type[App] = CMakeApp,
```
Comment (hfudev) [inline idf_build_apps/finder.py:31]: better to keep the name

```suggestion
    build_system: t.Union[t.Type[App], str] = CMakeApp,
```

**Review by hfudev — CHANGES_REQUESTED** [review]
LGTM! Thank you for your PR @xd-ydchen

File: idf_build_apps/finder.py (hunk 2 above)
Comment (xd-ydchen) [reply idf_build_apps/finder.py:31]: I move `build_system -> app_cls` to method `find_apps`. 
It seems that this method `_get_apps_from_path` is an internal method. 
If I keep this method compatible with str parameters, I have to check the parameter type before the log in Line 156. Do you think it is important to let this method also be compatible with str parameter?

File: CHANGELOG.md (hunk 1 above)
Comment (xd-ydchen) [reply CHANGELOG.md]: removed changes in this file

File: idf_build_apps/finder.py (hunk 2 above)
Comment (hfudev) [reply idf_build_apps/finder.py:31]: yes you're right. I agree to keep as what it is right now

**Comment by hfudev** [comment]
Thank you @xd-ydchen !

**Pull request merged**